### PR TITLE
Merge 1.0 maint - TW's version

### PR DESCRIPTION
**Pull request by ThomasWaldmann**

### CODE REVIEW

File: src/borg/key.py [hunk 1]
```diff
@@ -313,6 +315,9 @@ class ID_HMAC_SHA_256:
     def id_hash(self, data):
         return hmac_sha256(self.id_key, data)
 
+    def _tam_key(self, salt, context):
+        return salt + context
+
```
Comment (enkore) [inline src/borg/key.py]: git tried to sneak than in in my merge as well.

File: src/borg/helpers.py [hunk 2]
```diff
@@ -249,7 +255,7 @@ def write(self):
         self.tam_verified = True
         data = self.key.pack_and_authenticate_metadata(manifest.as_dict())
         self.id = self.key.id_hash(data)
-        self.repository.put(self.MANIFEST_ID, self.key.encrypt(Chunk(data, compression={'name': 'none'})))
+        self.repository.put(self.MANIFEST_ID, self.key.encrypt(Chunk(data), none_compression=True))
```
Comment (enkore) [inline src/borg/helpers.py]: I'd leave it with the existing compression support, that way no key.py changes are needed at all and no extra API.

File: src/borg/testsuite/archiver.py [hunk 3]
```diff
@@ -2358,6 +2357,82 @@ def test_disable2(self):
...
+        self.spoof_manifest(repository)
+        assert not self.cmd('list', self.repository_location)
+
+    def test_disable2(self):
+        self.cmd('init', '--encryption=repokey', self.repository_location)
+        self.create_src_archive('archive1234')
+        repository = Repository(self.repository_path, exclusive=True)
+        self.spoof_manifest(repository)
+        self.cmd('upgrade', '--disable-tam', self.repository_location)
+        assert not self.cmd('list', self.repository_location)
```
Comment (enkore) [inline src/borg/testsuite/archiver.py]: duplicated class

File: src/borg/helpers.py (hunk 2 above)
Comment (ThomasWaldmann) [inline src/borg/helpers.py]: not in enkore's merge, maybe solved differently there.

File: src/borg/archive.py [hunk 4]
```diff
@@ -1360,16 +1360,22 @@ def valid_item(obj):
             sort_by = sort_by.split(',')
             if any((first, last, prefix)):
                 archive_infos = self.manifest.archives.list(sort_by=sort_by, prefix=prefix, first=first, last=last)
+                if prefix and not archive_infos:
+                    logger.warning('--prefix %s does not match any archives', prefix)
+                if first and len(archive_infos) < first:
+                    logger.warning('--first %d archives: only found %d archives', first, len(archive_infos))
+                if last and len(archive_infos) < last:
+                    logger.warning('--last %d archives: only found %d archives', last, len(archive_infos))
```
Comment (enkore) [inline src/borg/archive.py]: I'd say - I pick this change over in my branch and we merge that one?

File: src/borg/key.py [hunk 5]
```diff
@@ -156,7 +156,7 @@ def compress(self, chunk):
         data = compressor.compress(data)
         return Chunk(data, **meta)
 
-    def encrypt(self, chunk):
+    def encrypt(self, chunk, none_compression=False):
```
Comment (ThomasWaldmann) [inline src/borg/key.py]: not in enkore's merge

File: src/borg/key.py [hunk 6]
```diff
@@ -253,7 +253,9 @@ def detect(cls, repository, manifest_data):
     def id_hash(self, data):
         return sha256(data).digest()
 
-    def encrypt(self, chunk):
+    def encrypt(self, chunk, none_compression=False):
+        if none_compression:
+            chunk.meta['compress'] = CompressionSpec('none')
```
Comment (ThomasWaldmann) [inline src/borg/key.py]: not in enkore's merge

File: src/borg/key.py (hunk 1 above)
Comment (ThomasWaldmann) [inline src/borg/key.py]: not in enkore's merge

File: src/borg/key.py [hunk 7]
```diff
@@ -331,7 +336,9 @@ class AESKeyBase(KeyBase):
 
     MAC = hmac_sha256
 
-    def encrypt(self, chunk):
+    def encrypt(self, chunk, none_compression=False):
+        if none_compression:
+            chunk.meta['compress'] = CompressionSpec('none')
```
Comment (ThomasWaldmann) [inline src/borg/key.py]: not in enkore's merge

File: src/borg/repository.py [hunk 8]
```diff
@@ -11,8 +11,8 @@
 
 import msgpack
 
-import logging
-logger = logging.getLogger(__name__)
+from .logger import create_logger
+logger = create_logger()
```
Comment (ThomasWaldmann) [inline src/borg/repository.py]: not in enkore's merge.

File: src/borg/key.py (hunk 1 above)
Comment (enkore) [inline src/borg/key.py]: https://github.com/borgbackup/borg/pull/2022#discussion_r95820346

File: src/borg/testsuite/archive.py [hunk 9]
```diff
@@ -30,8 +30,8 @@ def test_stats_basic(stats):
     assert stats.usize == 10
 
 
-def tests_stats_progress(stats, columns=80):
-    os.environ['COLUMNS'] = str(columns)
+def tests_stats_progress(stats, monkeypatch, columns=80):
+    monkeypatch.setenv('COLUMNS', str(columns))
```
Comment (ThomasWaldmann) [inline src/borg/testsuite/archive.py]: not in enkore's merge.

File: src/borg/testsuite/archiver.py [hunk 10]
```diff
@@ -2358,6 +2357,82 @@ def test_disable2(self):
         assert not self.cmd('list', self.repository_location)
 
 
+class ManifestAuthenticationTest(ArchiverTestCaseBase):
```
Comment (ThomasWaldmann) [inline src/borg/testsuite/archiver.py]: not in enkore's merge.
hmm, did i duplicate something here?

File: src/borg/upgrader.py [hunk 11]
```diff
@@ -3,13 +3,12 @@
...
 
-import logging
-logger = logging.getLogger(__name__)
-
 from .constants import REPOSITORY_README
 from .helpers import get_home_dir, get_keys_dir, get_cache_dir
 from .helpers import ProgressIndicatorPercent
 from .key import KeyfileKey, KeyfileNotFoundError
+from .logger import create_logger
+logger = create_logger()
```
Comment (ThomasWaldmann) [inline src/borg/upgrader.py]: not in enkore's merge.

File: src/borg/key.py (hunk 1 above)
Comment (ThomasWaldmann) [inline src/borg/key.py]: it was not even in a conflict. good that you verified it.

File: src/borg/helpers.py (hunk 2 above)
Comment (ThomasWaldmann) [inline src/borg/helpers.py]: ok

File: src/borg/archive.py (hunk 4 above)
Comment (ThomasWaldmann) [inline src/borg/archive.py]: yup, merge after taking all the good pieces.

**Comment by enkore** [comment]
We did some double work here, but "kinda worth it" finding some merge issues + enhancements in these.

**Comment by ThomasWaldmann** [comment]
exactly. :)

**Pull request closed without merging**